### DISCUSSION TITLE
removal of 'html_login'

### DISCUSSION THREAD

**Discussion by snooptodd** [sabnzbd/sabnzbd#2197] [General]
I don't remember why i changed it. I definitely don't need it disabled any more, if i ever did.

**Comment by thezoggy** [comment]
setting: config > specials > html_login  

html_login by default is enabled. 
if its not enabled, the login prompt is the old school basic auth popup instead.

majority of people want the html login page so they can use pw managers and overall just nicer to deal with

**Comment by naydichev** [comment]
I use nginx as a proxy and set the authentication header to bypass authentication. I think I had to do this because I was accessing sabnzbd from an 'external' IP, and it was required. I may be misremembering the reason, but I just wanted authentication to be disabled and this was the best way to 'disable' it.

**Reply by sanderjo** [reply]
Can you go to Config -> General (http://127.0.0.1:8080/sabnzbd/config/general/), and at External internet access (dropdown) select "Full Web Webinterface, Only external access requires login"
... and turn off HTML_login, and then try again?

**Reply by naydichev** [reply]
I made the external access change, but I can't seem to disable html_login. It is unchecked, but has an asterisk by it in the special config page and says 'on'

**Reply by Safihre** [reply]
The default is enabled, the Asterix indicates that you have changed it from the default by unchecking it. So check it again and then see if it still works as desired.

**Reply by naydichev** [reply]
got it. I checked it, the asterisk went away and it still seems to work fine 👍

**Comment by schabau** [comment]
Hello,

I'm pretty sure I never disabled the setting. I didn't even know the setting until the update to 3.6.0. I always had the login popup window in all past versions of Sabnzbd.

But now with the html_login I like it much better.

Many greetings
Schabau

**Comment by jaybeedk** [comment]
I probably had html_login disabled because early on it was the easiest way to have usr/pwd prefilled for me.
The reverse is now true, so I have enabled it again (default setting)

**Reply by jaybeedk** [reply]
@Safihre - ~~however it would be cool with a "Remember this device" checkbox, so you could skip login for longer (e.g. 90 days)~~
Never mind, there IS a "Remember me" checkbox (I had to check again).
I guess the session is lost when the server reboots?

**Reply by jaybeedk** [reply]
It would be nice, if the session could persist across server shutdowns (as mine is shut down each night). If not permanently, then for a period, e.g. 30 days

**Comment by koolsb** [comment]
I host sabnzbd behind a reverse proxy which also handles authentication for me. In order to not leave sabnzbd exposed on my local network I use basic auth that my reverse proxy automatically provides with the auth header, thus bypassing the sabnzbd login form yet still preventing unauthenticated users from accessing it.

**Reply by Safihre** [reply]
Hmm that sounds like a valid use case indeed. @sanderjo I don't think we can work around this, right?

**Reply by thezoggy** [reply]
I've seen plenty of people use reverse proxy and sab, and they just use the existing option "Full Web interface - Only external access requires login" and using local_range to know what to bypass (also people have to have correctly configured reverse proxy)

you using authelia? if so: https://docs.ibracorp.io/authelia/nginx/nginx-config-authelia

**Reply by sanderjo** [reply]
@Safihre I have zero experience with proxy authentication ....

**Comment by mirisbowring** [comment]
I am using sabnzbd behind a Authentik SSO. Authentik acts as a authentication proxy and passes the credentials as Basic auth.
Via Authentik i can grant users access to specific applications.

An alternative i could think of would be a custom authentication header but we could keep it as it ist 😊

**Comment by tapufd** [comment]
I have html_login set to off, because I use a password manager, and by this it automatically uses the creds from the password manager AND logs in automatically.
Else I still need to do a manual action to login... press the login-button, as the creds are auto-filled in, but the button I still need to press myself.

**Comment by Safihre** [comment]
Based on all your feedback, the `html_login` will be kept. Thanks all!
